src/style.re

```diff
@@ -1,5 +1,53 @@
 type t;
 
+type styleElement = (string, Js.Json.t);
+
+type px_pct =
+  | Px(float)
+  | Pct(float);
+
+let encode_px_pct = (value) =>
+  switch value {
+  | Px(px) => Encode.float(px)
+  | Pct(pct) => Encode.pct(pct)
+  };
+
+type px_auto =
+  | Px(float)
+  | Auto;
+
+let encode_px_auto = (value) =>
+  switch value {
+  | Px(value) => Encode.float(value)
+  | Auto => Encode.string("auto")
+  };
+
+type px_pct_animated_interpolated =
+  | Px(float)
+  | Pct(float)
+  | Animated(AnimatedRe.Value.t)
+  | Interpolated(AnimatedRe.Interpolation.t);
+
+let encode_px_pct_animated_interpolated = (value: px_pct_animated_interpolated) =>
+  switch value {
+  | Px(px) => Encode.float(px)
+  | Pct(pct) => Encode.pct(pct)
+  | Animated(value) => Encode.animatedValue(value)
+  | Interpolated(value) => Encode.interpolatedValue(value)
+  };
+
+type float_interpolated_animated =
+  | Float(float)
+  | Animated(AnimatedRe.Value.t)
+  | Interpolated(AnimatedRe.Interpolation.t);
+
+let encode_float_interpolated_animated = (value) =>
```

Comment:
I think we should stick to camelCase for these names since we mostly have camelCase in the codebase already and the convention in Reason is camelCase.

Reply:
I agree but this is not user facing and is easier to read for me than `type floatInterpolatedAnimated` vs `type float_interpolated_animated` and encode follows that. The underscore symbolises `or` here.